### ISSUE TITLE
dark mode for website

### ISSUE THREAD

**Issue by MehfoozurRehman** [expressjs/expressjs.com#1382]
would like to help here with making a dark theme for expressjs.com

**Comment by dougwilson** [comment]
Sounds awesome! I look forward to your contribution!

**Comment by hugoovf** [comment]
Just remove the background image every time u join a page of the website, or make a pr removing the file xd
![image](https://user-images.githubusercontent.com/69076992/212496121-d6cd658c-6a0b-4f19-9e2b-89922eb09383.png)

**Comment by tabishnaqvi1311** [comment]
Hello @MehfoozurRehman , I noticed that you opened this issue three months ago about adding dark mode. I'm interested in contributing to the project and was wondering if you are still working on this issue or if you would be willing to let me take a look at it? Please let me know if there's anything I can do to help.

**Comment by codesmith25103** [comment]
Hey @dougwilson, is this issue still open? If yes, can you please assign me this issue?

**Comment by faisalsaifii** [comment]
What's pending here?

**Comment by MehfoozurRehman** [comment]
Hey everyone,

Apologies for my absence last year due to studies. I'm comfortable with the MERN stack but not Ruby. However, I use Express.js daily and am eager to implement dark mode. Can someone guide me on running the project locally? I'm ready to contribute.

Thanks,

**Comment by MehfoozurRehman** [comment]
ok so last year I tried doing the local run according to repo readme and again today i tried but 

![image](https://github.com/expressjs/expressjs.com/assets/68416499/a1b4adda-1d25-4d82-994c-de4f095f7067)

its giving me this error for some reason to make sure it's not my local pc i am using GitHub code spaces.

![image](https://github.com/expressjs/expressjs.com/assets/68416499/3e30f555-358a-4f23-bbe8-09451e881843)
![image](https://github.com/expressjs/expressjs.com/assets/68416499/b8fcaeea-ecb9-4ef7-b9e3-32caaa03d026)

these is the result of first steps I am not really familiar with the ruby ecosystem so please let me know what i am doing wrong.

**Comment by MehfoozurRehman** [comment]
and i have read the issues created based on this one so what i am thinking once i have this running locally 

![image](https://github.com/expressjs/expressjs.com/assets/68416499/216aece2-b737-411b-bd6a-ffa41261d7c9)

the area where i have marked red i am thinking of adding 

![image](https://github.com/expressjs/expressjs.com/assets/68416499/0b03bead-bfa1-4c71-88eb-cb43351933c1)
![image](https://github.com/expressjs/expressjs.com/assets/68416499/a64cc897-d206-4704-b61c-2bfa7fa73c75)

these two icons first there can be some js to check the user device preference and adjust the theme according to that and if a user switches the theme, we save the preference in local storage. also i have read the concern about removing the background image from body i will make sure i pick the background color from that image for the light mode

**Comment by chrisdel101** [comment]
Hey Guys, was this still active? And/or still being worked on? I'm looking to make some contributions here.

**Comment by crandmck** [comment]
@chrisdel101 Please do....  I haven't delved into this issue, but if you can sort it out your contribs are welcome!

**Comment by chrisdel101** [comment]
okay, I'm looking into this and will plan to handle it.

**Comment by MehfoozurRehman** [comment]
guys can someone please help me run the repo locally

**Comment by hugoovf** [comment]
Hey, just read the [/README.md#local-setup](https://github.com/chrisdel101/expressjs.com?tab=readme-ov-file#local-setup)

**Comment by MehfoozurRehman** [comment]
> ok so last year I tried doing the local run according to repo readme and again today i tried but
> 
> ![image](https://private-user-images.githubusercontent.com/68416499/303423737-a1b4adda-1d25-4d82-994c-de4f095f7067.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI5MDM0NzQsIm5iZiI6MTcxMjkwMzE3NCwicGF0aCI6Ii82ODQxNjQ5OS8zMDM0MjM3MzctYTFiNGFkZGEtMWQyNS00ZDgyLTk5NGMtZGU0ZjA5NWY3MDY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDEyVDA2MjYxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4N2Y3ZjVlMTNmMGY4NTI4ODk0OGFkZDFhYTcwMzgxZWM2OTIyZDk0ZjFmOWRhYzkwNjJlM2RhM2UzYjI3YzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vTJVsbLxcjpOJ7E9b8HOqL6esbIMWfHH8iJWIxDGG-4)
> 
> its giving me this error for some reason to make sure it's not my local pc i am using GitHub code spaces.
> 
> ![image](https://private-user-images.githubusercontent.com/68416499/303424128-3e30f555-358a-4f23-bbe8-09451e881843.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI5MDM0NzQsIm5iZiI6MTcxMjkwMzE3NCwicGF0aCI6Ii82ODQxNjQ5OS8zMDM0MjQxMjgtM2UzMGY1NTUtMzU4YS00ZjIzLWJiZTgtMDk0NTFlODgxODQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDEyVDA2MjYxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmYTQxOGZkM2IwMzY3NjA5MWY0MjY4NWQ1ZTI3YjcyZjZlZTIxNzQxYmIwYzhjZjdhZmI4ZTJlZWZlNTBhMWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oOOFD3HYYT-9XxHQWnjjmDmX4GhzeJWhXu-PS670SEk) ![image](https://private-user-images.githubusercontent.com/68416499/303424254-b8fcaeea-ecb9-4ef7-b9e3-32caaa03d026.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI5MDM0NzQsIm5iZiI6MTcxMjkwMzE3NCwicGF0aCI6Ii82ODQxNjQ5OS8zMDM0MjQyNTQtYjhmY2FlZWEtZWNiOS00ZWY3LWI5ZTMtMzJjYWFhMDNkMDI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDEyVDA2MjYxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE4NWVmMjNhMjM1ODI0NWViODI2OTliZGYyMWE4YThhNzg4Y2ZiNzM2OTkzODdkYTA1MmU4MGE1YjVhMGFkZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5a6GLQN0-x7yCAJskxEYj6VvZE965blvUD7L2_8OJsU)
> 
> these is the result of first steps I am not really familiar with the ruby ecosystem so please let me know what i am doing wrong.

i have sent this before the instructions in readme are not working for me please can you help here

**Comment by MehfoozurRehman** [comment]
i have tried on github codespaces and my local computer its the same

**Comment by chrisdel101** [comment]
@MehfoozurRehman I've already started working on this, see above, and have made some significant progress. Prob not worth you also working on at this point.

If you are set on it, I am making this as a "theme", meaning there could be numerous themes. So after I complete PR this you could add another "theme". Like GH has multiple dark themes?

**Comment by MehfoozurRehman** [comment]
@chrisdel101, it's okay. Please continue working on this. I was under the impression that this issue was abandoned, but if you can, please complete the pull request.

**Comment by chrisdel101** [comment]
Re: your setup, the images above are not showing up for me. After you install Jekyll and run the serve command what happens?

**Comment by MehfoozurRehman** [comment]
![image](https://github.com/expressjs/expressjs.com/assets/68416499/6dffd96f-8848-4ba5-ab09-023ba9f04214)
this is what I am getting
I followed the readme file here

**Comment by chrisdel101** [comment]
Huh, yeah I have no idea why you are getting that. Seems like something is not intstalled correctly, or a gem is missing. I assume you did all these steps first:

1.  [https://jekyllrb.com/docs/installation/](https://jekyllrb.com/docs/installation/)
2.  [https://jekyllrb.com/docs/](https://jekyllrb.com/docs/)

If something failed in the first two steps then the server will fail. Maybe try the example in the docs they give and see you can run the basic example there. You should have something like this before it will run
```
ruby-v
ruby 3.3.0
```
```
jekyll -v
jekyll 4.3.3
```
Install extra gem from [`/README.md#local-setup`](https://github.com/chrisdel101/expressjs.com?tab=readme-ov-file#local-setup)
`gem install jekyll-redirect-from`

Then after cloning/forking the expressJS repo, this is what I see:
```
bundle install
Fetching gem metadata from https://rubygems.org/..........
Resolving dependencies...
Fetching nokogiri 1.16.4 (x86_64-darwin)
Installing nokogiri 1.16.4 (x86_64-darwin)
Bundle complete! 3 Gemfile dependencies, 96 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
```
```
bundle exec jekyll serve
/usr/local/lib/ruby/gems/3.3.0/gems/jekyll-3.9.5/lib/jekyll.rb:28: warning: csv was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add csv to your Gemfile or gemspec. Also contact author of jekyll-3.9.5 to add csv into its gemspec.
Configuration file: /Users/myComputer/code_work/jekyll/expressjs_copy/_config.yml
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
            Source: /Users/myComputer/code_work/jekyll/expressjs_copy
       Destination: /Users/myComputer/code_work/jekyll/expressjs_copy/_site
 Incremental build: enabled
      Generating... 
                    done in 15.486 seconds.
 Auto-regeneration: enabled for '/Users/myComputer/code_work/jekyll/expressjs_copy'
LiveReload address: http://127.0.0.1:35729
    Server address: http://127.0.0.1:4000
  Server running... press ctrl-c to stop.
        LiveReload: Browser connected
```

**Comment by MehfoozurRehman** [comment]
![image](https://github.com/expressjs/expressjs.com/assets/68416499/53d37425-0079-487f-869d-a80bf7a979ad)
![image](https://github.com/expressjs/expressjs.com/assets/68416499/2b2b59a5-5932-4d22-8994-9e83ed7337fe)
![image](https://github.com/expressjs/expressjs.com/assets/68416499/5668979f-8bf6-47bd-9992-57218c40efc4)
![image](https://github.com/expressjs/expressjs.com/assets/68416499/62c16e7f-513a-4547-8922-49b606f7c529)
![image](https://github.com/expressjs/expressjs.com/assets/68416499/8806c7b4-3839-4972-b470-66d28e8ee812)
yes so I have done the steps mentioned above It is saying the same thing for me just my ruby and Jakyell versions are different I am using GitHub code spaces for this these are the versions installed on that by default

**Comment by MehfoozurRehman** [comment]
![image](https://github.com/expressjs/expressjs.com/assets/68416499/e79367c9-4284-43e8-a2d7-a92ac577437f)
ok so I was trying a few things i didn't know how to update the ruby version I changed the codespaces settings to use the beta docker image and gem update --system 3.5.9 updated the ruby using this. I don't know which one fixed it but now it's working
![image](https://github.com/expressjs/expressjs.com/assets/68416499/690e6695-e60c-4c9f-808f-bdf441bfbb7c)
anyways appreciate the help please let me where you are in completing this and if you need any help with this issue

**Comment by chrisdel101** [comment]
Hey guys, I ran into a problem. 
This is the `_layouts/home.html` which is used for the main `index.md.` It pulls a JS file I added from the JS folder. It needs to run at the end of the page, and it works.

```
<html lang="{{ page.lang }}">
   <body class="en-doc">
   <script data-cfasync="false" src="../js/theme.js"></script>
   </body>
</html>
```

This is the `_layouts/page.html` which runs most of the other pages, for example `installing.md`.  It pulls a JS file I added to the same folder `_layouts` just for test purposes to show the problem.
```
<html lang="{{ page.lang }}">
   <body class="en-doc">
   <script data-cfasync="false" src="test.js"></script>
   </body>
</html>
```
I get `ERROR 'en/starter/test.js' not found.` It is ignoring the `src`, even absolute paths, and pulling for path where the markdown file is located `en/starter/markdown.md`. How do I stop this? Why is this happening?

The overall point is all the pages need that `theme.js` file


Update-- Status: working now.
